model-optimizer/extensions/ops/If.py

```diff
@@ -0,0 +1,319 @@
+
+        :param if_node: The If node to update output ports and shapes
+        :return: None
+        """
+
+        then_outputs = [node for node in if_node.then_graph.get_op_nodes() if node.has('output_id')]
+        else_outputs = [node for node in if_node.else_graph.get_op_nodes() if node.has('output_id')]
+        outputs_mapping = {}
+        outputs_number = len(if_node.out_ports())
+
+        if outputs_number == 0 and len(if_node.out_ports(True)) != 0:
+            # shape inference for asserts
+            for node in if_node.out_nodes(True).values():
+                node.shape = int64_array([])
+            return
+
+        for port_id in if_node.out_ports().keys():
+            outputs_mapping[port_id] = {}
+        port_ids = outputs_mapping.keys()
+
+        then_contains_fake_outputs=False
+        else_contains_fake_outputs=False
+
+        for then_output_node in then_outputs:
+            assert then_output_node.soft_get('type') == 'Result'
+            port_id = then_output_node['output_id']
+            assert port_id in port_ids, 'Incorrect mapping then_graph outputs with \"{0}\" outputs! ' \
+                                        'Can\'t find port with ID \"{1}\" in \'If\' operation.' \
+                .format(then_output_node.name, port_id)
+            outputs_mapping[port_id]['then_graph'] = then_output_node
+            then_shape = then_output_node.in_port(0).data.get_shape()
+
+            then_contains_fake_outputs = then_contains_fake_outputs or (len(then_shape)==1 and then_shape[0]==0)
+
+        for else_output_node in else_outputs:
+            assert else_output_node.soft_get('type') == 'Result'
+            port_id = else_output_node['output_id']
+            assert port_id in port_ids, 'Incorrect mapping then_graph outputs with \"{0}\" outputs! ' \
+                                        'Can\'t find port with ID \"{1}\" in \'If\' operation.' \
+                .format(else_output_node.name, port_id)
+            outputs_mapping[port_id]['else_graph'] = else_output_node
+            else_shape = else_output_node.in_port(0).data.get_shape()
+            else_contains_fake_outputs = else_contains_fake_outputs or (len(else_shape)==1 and else_shape[0]==0)
+
+        use_then_shape = else_contains_fake_outputs or not then_contains_fake_outputs
+        for port_id in outputs_mapping:
+            then_else_nodes = outputs_mapping[port_id]
+            assert 'then_graph' in then_else_nodes.keys(), 'then_graph does not connect with If.out_port[{0}] ' \
+                                                           'in \"{1}\" node!'.format(port_id, if_node.name)
+            assert 'else_graph' in then_else_nodes.keys(), 'else_graph does not connect with If.out_port[{0}] ' \
+                                                           'in \"{1}\" node!'.format(port_id, if_node.name)
+
+            then_shape = then_else_nodes['then_graph'].in_port(0).get_connection().data.get_shape()
```

Comment:
I think that you can remove `.get_connection()` in this and the next line.

Reply:
done